### DISCUSSION TITLE
Mongoose Types.ObjectId is missing in Prisma

### DISCUSSION THREAD

**Discussion by raviroshanmehta** [prisma/prisma#18568] [Q&A] [answered]
## Problem
string to object id conversion not available like mongoose ?

var mongoose = require('mongoose');
var id = mongoose.Types.ObjectId('4edd40c86762e0fb12000003');

If not available, do we have any option to achieve it ?

**Comment by ludralph** [comment, marked as answer]
Hi @raviroshanmehta 👋 

You can use the official MongoDB [BSON](https://www.npmjs.com/package/bson) package to generate an ObjectId in Prisma.

Given this schema
```prisma
model Users {
  id    String @id @map("_id")
  email String @unique
  name  String?
}
```
Then we could use the library as shown below

```js
const { ObjectId } = require('bson');
const res = await prisma.users.create({
  data:{
    id: new ObjectId().toString(),
    email: "test@gmail.com",
    name: "test name"
  }
})
```
This give an output of 
```js
{
  id: '642540e1ff673bac6e870194',
  email: 'test@gmail.com',
  name: 'test name'
}
```
To convert a string to an objectId, you can simply pass the string to new ObjectId() method
```js
console.log(new ObjectId('4edd40c86762e0fb12000003'))
```
This returns
```
new ObjectId("4edd40c86762e0fb12000003")
```
You can take a look at this [issue](https://github.com/prisma/prisma/issues/6718) as well

**Comment by raviroshanmehta** [comment]
This bson module is working fine. But this solution is not working for
prisma aggregateRaw queries. match pipeline of prisma is not returning
results.

However we have $toObjectId in aggregations but isn't possible match in
pipeline will support ObjectId directly in query.

**Comment by raviroshanmehta** [comment]
Found solution at https://github.com/prisma/prisma/issues/6718
